yt/data_objects/index_subobjects/stretched_grid.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+from yt.geometry.selection_routines import convert_mask_to_indices
+
+from .grid_patch import AMRGridPatch
+
+
+class StretchedGrid(AMRGridPatch):
+    def __init__(self, id, cell_widths, filename=None, index=None):
+        self.cell_widths = [np.array(_) for _ in cell_widths]
+        super().__init__(id, filename, index)
+
+    def _check_consistency(self):
+        computed_right_edge = self.LeftEdge + [_.sum() for _ in self.cell_widths]
+        assert (computed_right_edge == self.RightEdge).all()
+
+    def _get_selector_mask(self, selector):
+        if self._cache_mask and hash(selector) == self._last_selector_id:
```

Comment:
Maybe?  Came right out of `grid_patch.py`

Reply:
actually it's a boolean, and it's currently _never_ set to anything else than `True`, so I propose we remove it instead of cargo cutting it any further (https://github.com/yt-project/yt/pull/4048)

Reply:
I was incorrect, `_cache_mask` is a boolean, so, nothing to do here